test/fixtures/test-runner/output/test-timeout-flag.js

```diff
@@ -0,0 +1,25 @@
+// Flags: --test-timeout=20
+'use strict';
+const { describe, test, afterEach } = require('node:test');
+const { setTimeout } = require('node:timers/promises');
+
+let timeout;
+
+describe('--test-timeout is set to 20ms', () => {
+  test('should timeout after 20ms', async () => {
+    timeout = await setTimeout(2000);
+  });
+  test('should timeout after 5ms', { timeout: 5 }, async () => {
+    timeout = await setTimeout(2000);
+  });
+  test('should not timeout', { timeout: 5000 }, async () => {
+    await setTimeout(200);
+  });
+  test('should pass', async () => {});
+
+  afterEach(() => {
+    if (timeout) {
```

Comment:
Does it work to unconditionally `unref()` all of the timers? If so, we can get rid of this hook.

Reply:
sorry I didn't get this one, would you mind explaining a bit more in detail?

Reply:
I think I've made a big mistake as well - I am trying to assign return value from the `setTimeout promise` 😞

Reply:
Ahh I didn't know this, TIL 🙏 yes, it works with `{ ref: false }` thanks!